specs/projectreuniondeployment/projectreuniondeployment.md

```diff
@@ -0,0 +1,138 @@
+The API addresses #1 by a simple query of the version they seek and a possible check for OS
+updates. #2 has two possible answers. First is direct resolution by the developer by acquiring and
+installing the packages if possible. Some programs will not have permission or motivation to do
+this, so the second answer is providing a link up to the user where the user can take action to
+install the package. The API will address this scenario by providing an answer as to what is
+required, where to get it, and where to go for instruction.
+
+# Examples
+
+## GetProjectReunionStatus
+
+This is the means by which an app or program can call to verify Project Reunion is present to a
+minimum version. It should be called once after activation before trying to use Project Reunion
+functionality. This method returns a ProjectReunionStatus object that describes the current
+deployment state of Project Reunion.
+
+```C++
+Windows::ApplicationModel::PackageVersion version{1, 0, 0, 0};
+auto status =
+Microsoft::Management::Deployment::ProjectReunionDeployment::GetProjectReunionStatus(version);
+if (!status.IsOK())
+{
+    ...
+}
+```
+
+## GetProjectReunionPackageLink
+
+Provides a direct install link to deploy the latest version of ProjectReunion. If no OS Updates are
+required and the program or app has permissions to deploy packages, this can directly resolve the
+missing dependency without burdening the user.
+
+```C++
+Windows::Foundation::Uri packageLink = Microsoft::Management::Deployment::ProjectReunionDeployment::GetProjectReunionPackageLink();
+```
+
+## GetProjectReunionInstallInformationLink
+
+Provides a user-friendly link to acquire the latest version of Project Reunion and any OS updates
+that may be required. This can be used to guide the user to where they can resolve the missing
+dependency.
+
+```C++
+Windows::Foundation::Uri installInformationLink =
+Microsoft::Management::Deployment::ProjectReunionDeployment::GetProjectReunionInstallInformationLink();
+```
+
+# API Details
+
+```c#
+namespace Microsoft.Management.Deployment
+{
+    /// Represents the current Deployment status of Project Reunion
+    runtimeclass ProjectReunionDeploymentStatus
+    {
+        /// Returns True if Project Reunion is the desired Version and no updates or packages
+        /// are required.
+        Boolean IsOK { get; };
+
+        // Returns True if Windows Updates are required for Project Reunion.
+        Boolean WindowsUpdatesRequired { get; };
+
+        // Returns True if a Package install is required for Project Reunion.
+        Boolean PackageInstallRequired { get; };
+    }
+
+    /// Used to query deployment information for Project Reunion
+    static runtimeclass ProjectReunionDeployment
+    {
+        /// Returns the current deployment status of Project Reunion and whether it meets the 
+        /// minimum package version specified.
+        static ProjectReunionDeploymentStatus GetStatus(Windows.ApplicationModel.PackageVersion packageVersion);
```

Comment:
Do you also need an optional 'architecture' parameter?

Does ProjectReunionFramework.msix come in per-processor architecture flavors and there can be >1 registered to a user? e.g. on an x64 machine can I have ProjectReunionFramework-x64.msix and ProjectReunionFramework-x86.msix?

Will ProjectReunionMain come in multiple flavors e.g. ProjectReunionMainx64 and ProjectReunionMainx86?

If there's multiple possible architecture flavors registered to a user, do you check them all or only the ones I need, e.g. if I'm an x86 app do I care about Fwk-x86 *and* Fwk-x64? About Main-x86 *and* Main-x64?

Reply:
It will follow normal deployment applicability, which is on x64 machines would require and check both x86 and x64 frameworks are present and of the specified min version, which means an architecture parameter is not required. That also seems incorrect to me for a developer to specify architecture, when we can easily determine that and do the right thing.